app/controllers/documents/DocumentReferenceNumberController.scala

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.documents
+
+import controllers.actions._
+import forms.DocumentReferenceNumberFormProvider
+import models.requests.MandatoryDataRequest
+import models.{ArrivalId, CheckMode, Index, Mode, NormalMode}
+import navigation.Navigator
+import pages.documents.DocumentReferenceNumberPage
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents, Result}
+import repositories.SessionRepository
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
+import viewModels.documents.DocumentReferenceNumberViewModel.DocumentReferenceNumberViewModelProvider
+import views.html.documents.DocumentReferenceNumberView
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class DocumentReferenceNumberController @Inject() (
+  override val messagesApi: MessagesApi,
+  sessionRepository: SessionRepository,
+  actions: Actions,
+  navigator: Navigator,
+  formProvider: DocumentReferenceNumberFormProvider,
+  val controllerComponents: MessagesControllerComponents,
+  view: DocumentReferenceNumberView,
+  viewModelProvider: DocumentReferenceNumberViewModelProvider
+)(implicit ec: ExecutionContext)
+    extends FrontendBaseController
+    with I18nSupport {
+
+  private def prefix(mode: Mode): String = mode match {
+    case NormalMode => "document.referenceNumber.NormalMode"
+    case CheckMode  => "document.referenceNumber.CheckMode"
+  }
```

Comment:
Could this not be retrieved from the view model?

Reply:
I just need it in the form, so is it okay to move the logic over into the form @jackheslop96?